src/StaticWebAssetsSdk/Targets/Microsoft.NET.Sdk.StaticWebAssets.JSModules.targets

```diff
@@ -389,7 +393,7 @@ Copyright (c) .NET Foundation. All rights reserved.
   </Target>
 
   <!-- JS file modules -->
-  <Target Name="ResolveJSModuleStaticWebAssets" DependsOnTargets="ResolveProjectStaticWebAssets">
+  <Target Name="ResolveJSModuleStaticWebAssets" DependsOnTargets="ResolveProjectStaticWebAssets;ResolveRazorComponentInputs;ResolveRazorGenerateInputs">
```

Comment:
Looks like this change was missing in the original PR. Without this, the faulty behavior still occurs. Verified manually that this fixes the issue.

We'll want to bring this into `main` as well.

@javiercn FYI